Source/Tools/LinuxEmulation/LinuxSyscalls/FileManagement.cpp

```diff
@@ -258,16 +237,14 @@ FileManager::FileManager(FEXCore::Context::Context* ctx)
   for (const auto& Path : ConfigPaths) {
     fextl::vector<char> FileData;
     if (FEXCore::FileLoading::LoadFile(FileData, Path)) {
-      JSON::JsonAllocator Pool {
-        .PoolObject =
-          {
-            .init = JSON::PoolInit,
-            .alloc = JSON::PoolAlloc,
-          },
-      };
+      FEX::JSON::JsonAllocator Pool = FEX::JSON::GetPool();
 
       // If a thunks DB property exists then we pull in data from the thunks database
-      const json_t* json = json_createWithPool(&FileData.at(0), &Pool.PoolObject);
+      const json_t* json = FEX::JSON::CreateJSON(FileData, Pool);
```

Comment:
Isn't another issue here that `FileData` won't be null-terminated? One of the other callees explicitly uses `push_back(0)` to add a terminator.

Instead, we can just change all `FileData`s to `fextl::string` instead (and drop the explicit `push_back`) and change `&FileData.at(0)` to `FileData.c_str()`. This will avoid the need for a wrapper function too, since the pointer [will always be valid](https://en.cppreference.com/w/cpp/string/basic_string/data).

Reply:
The json library doesn't care about null terminator, so I removed that extraneous push_back instead.